docs/gomod.md

```diff
@@ -0,0 +1,196 @@
+# gomod
+
+<https://go.dev/ref/mod>
+
+* Overview [in the README][readme-gomod]
+* [Specifying modules to process](#specifying-modules-to-process)
+* [Using fetched dependencies](#using-fetched-dependencies)
+* [gomod flags](#gomod-flags)
+* [Vendoring](#vendoring)
+* [Understanding reported dependencies](#understanding-reported-dependencies)
+
+## Specifying modules to process
```

Comment:
Specifying modules to process => `--package ...`

This illustrates my point that the current names can be confusing at times.

Reply:
From the little bit of googling I did while writing this,

Package managers that use "package":
* [npm](https://docs.npmjs.com/about-packages-and-modules#about-packages)
* [yarn](https://classic.yarnpkg.com/en/docs/creating-a-package)
* [cargo](https://doc.rust-lang.org/cargo/appendix/glossary.html#package) (Rust)

Other names:
* Go uses "module"
* Python uses ["project"](https://packaging.python.org/en/latest/glossary/#term-Project) ("package" would be a [distribution package](https://packaging.python.org/en/latest/glossary/#term-Distribution-Package) or an [import package](https://packaging.python.org/en/latest/glossary/#term-Import-Package))
* Ruby uses ["gem"](https://guides.rubygems.org/) with occasional mentions of "package"
* Lua uses ["rock"](https://github.com/luarocks/luarocks/wiki/Creating-a-rock) or "package"

Do you know what Java uses?

Reply:
(I would argue that Python, Ruby and Lua do fit within "package", only Go is confused)

Reply:
TL;DR most package managers admit they manage "packages", but some are hipsters and Go is a special snowflake :smile:

Reply:
Although "project" may also be a competitor, Python is the obvious one but also in Julia the package is defined in [Project.toml](https://pkgdocs.julialang.org/v1/toml-files/) and Cabal has a [project.cabal](https://cabal.readthedocs.io/en/stable/cabal-project.html)

Reply:
I'd argue that Java sees this in a similar way to how Go does.

A Java [package](https://docs.oracle.com/javase/tutorial/java/package/packages.html) is pretty much a "folder" within the Java part of a project, not something that you can distribute independently.

A .jar file is called a "package", but I think the meaning changes to "something that is in a distributable format". Maven calls its main file a Project Object Model, or POM, so maybe a "project" or "module" is the distributable part of it.

Reply:
I would say that, for us, a `--package` is a "path within a Git repository that can contains package manager information and can be independently processed".

Probably my initial source of confusion was seeing in the `request.json` a key called "dependencies" and other one called "packages" (a dependency is a package, right?), and seing packages in "packages" that don't fall within the description above.

I did like the disclaimer you added, though, so I would say it LGTM, we can further discuss naming in some other opportunity.